site/en/blog/improved-pwa-offline-detection/index.md

```diff
@@ -0,0 +1,156 @@
+**TL;DR:** Offline support has been part of the PWA installability criteria
+since the beginning. We are updating the offline detection logic to ensure a
+PWA actually provides an offline experience, and close the loophole used by
+some developers to meet the installability critiera. In the future,
+sites with empty `fetch` event handlers will no longer meet the criteria.
+!!!
+
+[Progressive Web Apps (PWAs)](https://web.dev/pwa/) are a pattern for
+building modern, installable applications using web technology for mobile and
+desktop devices.
+
+One of the criteria for building a modern web experience, and not
+coincidentally PWAs, is that the app must continue to work even if the device
+is offline.  That means no Chrome Dino screen if the user loses network
+access on their device!
+
+The goal of all PWA criteria checks is to help ensure users have a high
+quality, app competitive experience when browsing the web. Chrome performs
+checks against [PWA criteria][pwa-criteria] before enabling the install
+capability for a PWA.
+
+Only apps that fulfill all core
+[Progressive Web App installability criteria][pwa-criteria], including support
+for an offline mode, can be installed to the device from Chrome.
+
+## Current offline detection logic
+
+Verification of offline support has been part of the browser's installability
+checklist of a PWA for several years. However, the check was a simplified
+heuristic.
+
+Previously, the installability check passed if the page had a service worker
+installed that included a `fetch` event handler.  Chrome did not have the
+capability to simulate requests through the service worker, so a full check of
+correct offline behaviour was not possible.
+
+{% img src="image/0g2WvpbGRGdVs0aAPc6ObG7gkud2/qrDoFGCM2s8pBvM4n7Dv.png", alt="Diagram of service worker", width="800", height="385" %}
+
+That meant that Chrome did not have the ability to validate whether the `fetch`
+event handler returned a valid resource with HTTP 200 during the offline check.
+Chrome only checked whether the service worker actually has a `fetch` handler.
+
+Starting in Chrome 89, Chrome has the ability to run simulated offline requests
+through the service worker, allowing us to update the offline detection logic
+to better reflect actual offline support of the application.
+
+## Updated offline detection logic
+
+The updated offline detection logic checks:
+
+1) That there is a service worker installed for the page.
+2) That the installed service worker has a `fetch` event.
+3) **That the installed service worker `fetch` event returns an HTTP 200
+   status code (indicating a successful fetch) in simulated offline mode.**
+
+## What does this mean for developers?
+
+A properly implemented offline mode requires that the `fetch` handler return
+a resource with HTTP 200, so, if you've correctly implemented an offline mode
+for your Progressive Web App, this change to the offline detection logic will
+not require any action on your part.
```

Comment:
```suggestion
for your Progressive Web App, this change to offline detection requires no
action by you.
```

Reply:
Aside from following a best practice of omitting needless words, this makes a primary point of the article more direct, and hence easier to remember.